src/components/PasswordInput/PasswordInput.js

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled, { css } from 'react-emotion';
+
+import { InputWrapper } from '../InputWrapper';
+import { Input as StandardInput } from '../Input';
+import { SvgButton } from '../SvgButton';
+import State from '../State/State';
+import RevealIconSvg from './eye.svg';
+import HideIconSvg from './eye-off.svg';
+
+const Input = styled(StandardInput, { label: 'password-input__input' })(
+  ({ theme }) => `
+    padding-right: ${theme.spacings.peta};
+  `
+);
+
+const stylesIcon = css`
+  label: password-input__icon;
+  position: absolute;
+  top: 50%;
+  right: 12px;
```

Comment:
Are these also "spacings"?

Reply:
Oh snap. That's still from prototyping 😆

Reply:
How about an Input with placeholders for icons like `icon={<RevealIconSvg />}`? Since most of the use cases are Icon left/right aligned. Then you can make this styled Icon part of the input: 
```js
<IconedInput icon={isVisible ? <RevealIconSvg /> : <HideIcon />} iconAlign={left/right} />
```

You can use render props pattern in order to place the icons also if will (i.e., pass `className` for positioning to render prop).

Reply:
Here's the mock we came up with:

```javascript
import React from 'react'

const Container = styled('div')`
  position: relative;
`
const Input = styled('input')`
`

const IconInput = ({ childPosition, children }) => {
  const isLeftPositioned = childPosition === 'left'
  const className = css({
    top: 50%;
    transform: translateY(-50%);
    left: isLeftPositioned ? '12px' : 'initial';
    right: isLeftPositioned ? 'initial' : '12px';
  })

  return (
    <Container>
      <Input />
      {children({ className })}
    </Container>
  )
}

// Somewhere else

<IconInput>
  {({ className }) => <MyIconComponent {...{ className }} />}
</IconInput>
```